vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt

```diff
@@ -93,11 +100,37 @@ class CreateDirectRoomActivity : SimpleFragmentActivity() {
         viewModel.onEach(CreateDirectRoomViewState::createAndInviteState) {
             renderCreateAndInviteState(it)
         }
+
+        viewModel.observeViewEvents {
+            when (it) {
+                CreateDirectRoomViewEvents.InvalidCode -> {
+                    Toast.makeText(this, R.string.invalid_qr_code_uri, Toast.LENGTH_SHORT).show()
+                    finish()
+                }
+                CreateDirectRoomViewEvents.DmSelf -> {
+                    Toast.makeText(this, R.string.cannot_dm_self, Toast.LENGTH_SHORT).show()
+                    finish()
+                }
+            }
+        }
+
+        qrViewModel.observeViewEvents {
+            when (it) {
+                is QrCodeScannerEvents.CodeParsed  -> {
+                    viewModel.handle(CreateDirectRoomAction.QrScannedAction(it.result))
+                }
+                is QrCodeScannerEvents.ParseFailed -> {
+                    Toast.makeText(this, R.string.qr_code_not_scanned, Toast.LENGTH_SHORT).show()
+                    finish()
+                }
+            }
```

Comment:
here too.

Reply:
In this case we are not interested in some events, so I'll add `else - > Unit`